### PR TITLE
Skip filtration for targeted scans

### PR DESCRIPTION
**Pull request by rosecodym**
### Description:
There is a scenario in which results filtration is known to cause problems, and this PR disables it in that scenario. (It _should_ cause problems more generally, but lacking any concrete cases of that, I want to tread lightly.)

### Checklist:
* [x] Tests passing (`make test-community`)?
* [x] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -905,7 +905,15 @@ func (e *Engine) verificationOverlapWorker(ctx context.Context) {
 					detectorKeysWithResults[detector.Key] = detector
 				}
 
-				results = e.filterResults(ctx, detector, results)
+				// If results filtration eliminates a rotated secret, then that rotation will never be reported. This
+				// problem can theoretically occur for any scan, but we've only actually seen it in practice during
+				// targeted scans. (The reason for this discrepancy is unclear.) The simplest fix is therefore to
+				// disable filtration for targeted scans, but if you're here because this problem surfaced for a
+				// non-targeted scan then we'll have to solve it correctly.
+				if chunk.chunk.SecretID == 0 {
```
Comment (mcastorina) [inline pkg/engine/engine.go:913]: I know there are comments saying `SecretID != 0` for targeted scans, but what do you think about adding a little helper function to whatever type `chunk.chunk` is.

```go
func (c Chunk) TargetedScan() bool { return c.SecretID != 0 }
```

File: pkg/engine/engine.go [hunk 2]
```diff
@@ -1043,7 +1051,14 @@ func (e *Engine) detectChunk(ctx context.Context, data detectableChunk) {
...
 		}
 
-		results = e.filterResults(ctx, data.detector, results)
+		// If results filtration eliminates a rotated secret, then that rotation will never be reported. This problem
+		// can theoretically occur for any scan, but we've only actually seen it in practice during targeted scans. (The
+		// reason for this discrepancy is unclear.) The simplest fix is therefore to disable filtration for targeted
+		// scans, but if you're here because this problem surfaced for a non-targeted scan then we'll have to solve it
+		// correctly.
+		if data.chunk.SecretID == 0 {
+			results = e.filterResults(ctx, data.detector, results)
```
Comment (mcastorina) [inline pkg/engine/engine.go:1060]: We filter in two places? 😕

**Review by mcastorina — APPROVED** [review]

File: pkg/engine/engine.go (hunk 2 above)
Comment (rosecodym) [reply pkg/engine/engine.go:1060]: yeah the verification overlap prevention basically just duplicated a stage of the pipeline :(

File: pkg/engine/engine.go (hunk 1 above)
Comment (rosecodym) [reply pkg/engine/engine.go:913]: I like the sentiment, but I also hate that whether a scan is targeted is stored in _chunk_ data and I think I'd prefer to not double down on that.

**Pull request merged**